smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -21,6 +20,24 @@ contract MixinPurchase is
 {
   event RenewKeyPurchase(address indexed owner, uint newExpiration);
 
+  event GasRefunded(address indexed receiver, uint refundedAmount, address tokenAddress);
+
+  uint8 private _gasRefundPercentage = 0; // default to 0
+
+  /**
+  * @dev Set a percentage of the key price to be refunded to the sender on purchase
+  */
+  function setGasRefundPercentage(uint8 _percent) external onlyLockManagerOrBeneficiary {
+    _gasRefundPercentage = _percent;
+  }
+  
+  /**
+  * @dev Returns percentage be refunded to the sender on purchase
+  */
+  function gasRefundPercentage() external view onlyLockManagerOrBeneficiary returns (uint8 percentage) {
```

Comment:
I think `uint8` is too small! We need to go up to 10,000 (basis points). Also this should not have the `onlyLockManagerOrBeneficiary` modifier :)